src/internal/components/dnd-container/index.tsx

```diff
@@ -0,0 +1,164 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { DndContext, DraggableAttributes, DragOverlay } from '@dnd-kit/core';
+import { SyntheticListenerMap } from '@dnd-kit/core/dist/hooks/utilities';
+import { arrayMove, SortableContext, useSortable, verticalListSortingStrategy } from '@dnd-kit/sortable';
+import { CSS } from '@dnd-kit/utilities';
+
+import Portal from '../portal';
+import { ReorderAnnouncements, ReorderOptions } from './interfaces';
+import useDragAndDropReorder from './use-drag-and-drop-reorder';
+import useLiveAnnouncements from './use-live-announcements';
+
+import styles from './styles.css.js';
+
+export interface DndContainerProps<Option> extends ReorderOptions<Option> {
+  onChange: (sortedOption: readonly Option[]) => void;
+  disabled?: boolean;
+  renderOption: (props: RenderOptionProps<Option>) => React.ReactNode;
+  renderActiveOption: (props: RenderActiveOptionProps<Option>) => React.ReactNode;
+  i18nStrings: ReorderAnnouncements;
+  dragOverlayClassName?: string;
+  children?: (optionsContent: React.ReactNode) => React.ReactNode;
+}
+
+interface RenderOptionProps<Option> extends RenderActiveOptionProps<Option> {
+  ref: React.RefCallback<HTMLElement>;
+  style: React.CSSProperties;
+  isDragging: boolean;
+  isSorting: boolean;
+  attributes: DraggableAttributes;
+}
+
+interface RenderActiveOptionProps<Option> {
+  option: Option;
+  dragHandleAriaLabel?: string;
+  listeners?: SyntheticListenerMap;
+}
+
+export function DndContainer<Option>({
+  getId,
+  sortedOptions,
+  onChange,
+  disabled,
+  renderOption,
+  renderActiveOption,
+  i18nStrings,
+  dragOverlayClassName = styles['drag-overlay'],
+  children = content => content,
```

Comment:
Do we need this ability to customize the rendering in this way? So far it is used only in the newly added example dev page and it looks to me like the same could be achieve by rendering the DndContainer inside the `tbody` instead of it wrapping the table.

Reply:
I didn't use it in collection preference to not cause accidental regression. However, the reason it is done this way is to prevent DndContext to render the live region inside a semantic container like <ul> or <tbody>. There is already this issue in the collection prefs but it is not captured by AXE because the collection prefs is not open by default in the test page.